cartoframes/data/observatory/catalog/repository/repo_client.py

```diff
@@ -1,125 +1,83 @@
 import os
-import requests
-from urllib.parse import urlsplit, urlunsplit, SplitResult
-from .....auth import Credentials, get_default_credentials
+
+from carto.do_dataset import DODataset
+
+from .....auth import Credentials
 
 DEFAULT_USER = 'do-metadata'
-API_BASE_PATH = 'api/v4/data/observatory'
-REQUEST_VERIFY_SSL = True
 
 
 class RepoClient:
 
-    __instance = None
-
     def __init__(self):
-        self._user_credentials = None
+        self._do_dataset = None
+        default_credentials = Credentials(DEFAULT_USER)
+        default_auth_client = default_credentials.get_api_key_auth_client()
+        self._defautl_do_dataset = DODataset(auth_client=default_auth_client)
```

Comment:
Minor typo: `s/_defautl_do_dataset/_default_do_dataset/` (consistently repeated below)

Reply:
Aside and out of scope: not sure those default credentials `do-metadata` with `default_public` api key work with the DO API, but they should, and we should test it (if they don't, it shall be just a matter of removing a line in do api)

Reply:
I tested it and it works :)